app/assets/javascripts/date_range_picker.js.erb

```diff
@@ -2,7 +2,13 @@
 // All this logic will automatically be available in application.js.
 
 $(document).ready(function() {
-  var rails_right_now = '<%= Time.zone.now.strftime('%Y%m%d') %>';
+  const rails_right_now = (() => {
+    const today = new Date();
+    const padInt = (num, minimumIntegerDigits) => num.toLocaleString(undefined, { minimumIntegerDigits });
+
+    return `${today.getFullYear()}${padInt(today.getMonth(), 2)}${padInt(today.getDate(), 2)}`;
+  })();
```

Comment:
I think we can address this by removing `rails_right_now` entirely and replacing it with a variable within the function (to better ensure that it is being executed on every page load... I don't understand JS that much haha). So like this:
```    if ($('#filters_date_range').length > 0) {
      var now = moment();
      $('#filters_date_range').daterangepicker({
          linkedCalendars: false,
          autoApply: true,
          ranges: {
             'All Time': [moment().subtract(100, 'years'), moment(rails_right_now)],
             'Today': [moment(), moment(rails_right_now)],
             'Yesterday': [moment().subtract(1, 'days'), moment(rails_right_now).subtract(1, 'days')],
             'Last 7 Days': [moment().subtract(6, 'days'), moment(rails_right_now)],
             'Last 30 Days': [moment().subtract(29, 'days'), moment(rails_right_now)],
             'This Month': [moment().startOf('month'), moment(rails_right_now).endOf('month')],
             'Last Month': [moment().subtract(1, 'month').startOf('month'), moment(rails_right_now).subtract(1, 'month').endOf('month')],
             'This Year': [moment().startOf('year'), moment(rails_right_now).endOf('year')]
          }
      }, callback);

      start_date = $('#filters_date_range').data("initial-start-date");
      end_date = $('#filters_date_range').data("initial-end-date");

      $('#filters_date_range').data('daterangepicker').setStartDate(start_date);
      $('#filters_date_range').data('daterangepicker').setEndDate(end_date);
    };
```

Reply:
Yeah, that makes sense. I wasn't familiar with the `moment` API, but I should have check to see what it's equivalent of `now` is. :+1: